Description: Fix the check for parent directory access
 The previous check checked for the existence of ".."
 in the URI. This does not work correctly as ".." may
 appear in the package name or elsewhere, as can be
 seen in libpcre3_8.30..-2_i386.deb.
Author: Julian Andres Klode <jak@debian.org>

---
Bug-Debian: http://bugs.debian.org/665301

--- apt-cacher-ng-0.7.2.orig/source/job.cc
+++ apt-cacher-ng-0.7.2/source/job.cc
@@ -565,7 +565,7 @@ void job::PrepareDownload() {
 		m_bClientWants2Close = 0==strncasecmp(m_pReqHead->h[header::CONNECTION], "close", 5);
 
     // "clever" file system browsing attempt?
-	if(stmiss != sRawUriPath.find("..")
+	if(stmiss != sRawUriPath.find("../")
 			|| stmiss != sRawUriPath.find("/_actmp")
 			|| startsWithSz(sRawUriPath, "/_"))
 		goto report_notallowed;
